packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/DSL_DataSpace_PureProtocolProcessorPlugin.ts

```diff
@@ -206,14 +211,41 @@ export class DSL_DataSpace_PureProtocolProcessorPlugin
           if (elementProtocol.executables) {
             element.executables = elementProtocol.executables.map(
               (executableProtocol) => {
-                const executable = new DataSpaceExecutable();
-                executable.title = executableProtocol.title;
-                executable.description = executableProtocol.description;
-                executable.executable = context.resolveElement(
-                  executableProtocol.executable.path,
-                  false,
-                );
-                return executable;
+                if (
+                  executableProtocol instanceof V1_DataSpaceTemplateExecutable
+                ) {
+                  const executable = new DataSpaceExecutableTemplate();
+                  executable.title = executableProtocol.title;
+                  executable.description = executableProtocol.description;
+                  executable.query = V1_buildRawLambdaWithResolvedPaths(
+                    executableProtocol.query.parameters,
+                    executableProtocol.query.body,
+                    context,
+                  );
+                  if (executableProtocol.executionContextKey) {
+                    executable.executionContextKey =
+                      executableProtocol.executionContextKey;
+                  }
+                  return executable;
+                } else if (
+                  executableProtocol instanceof
+                  V1_DataSpacePackageableElementExecutable
+                ) {
+                  const executable =
+                    new DataSpacePackageableElementExecutable();
+                  executable.title = executableProtocol.title;
+                  executable.description = executableProtocol.description;
+                  executable.executable = context.resolveElement(
+                    executableProtocol.executable.path,
+                    false,
+                  );
+                  return executable;
+                } else {
+                  throw new UnsupportedOperationError(
```

Comment:
Did we have any cases before where the `executableProtocol` wasn't an instance of `V1_DataSpacePackageableElementExecutable` but we still built the `DataSpaceExecutable` instead of throwing an error? Would this change break any of those cases?

Reply:
`DataSpacePackageableElementExecutable` and `DataSpaceExecutableTemplate` extend abstract class `DataSpaceExecutable`. Before, we only had DataSpaceExecutable.

Since we make DataSpacePackageableElementExecutable as a default implementation when doing serialization and deserialization in Studio, it's backward-compatible.  I don't think we will fall into this else condition, but just in case there might be an issue, i created this exception.